File: my-games-backend/src/main/java/com/tc/mygames/domain/Game.java
package com.tc.mygames.domain;

import lombok.Data;
import lombok.Getter;
import lombok.Setter;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.io.Serializable;
import java.time.LocalDate;
import java.util.UUID;

@Data
@Getter
@Setter
@Document
public class Game implements Serializable {

    @Id
    private UUID id;

    private String title;

    private LocalDate completionDate;

    private String console;

    private LocalDate year;

    private String notes;

    private Boolean completed;

    public Game(String title, LocalDate completionDate, String console, LocalDate year, String notes, Boolean completed) {
        this.id = UUID.randomUUID();
        this.title = title;
        this.completionDate = completionDate;
        this.console = console;
        this.year = year;
        this.notes = notes;
        this.completed = completed;
    }
}

File: my-games-frontend/src/model/Game.ts
import { DateTime } from 'luxon';

class Game {
  public id?: String;

  public title?: String;

  public completionDate?: DateTime;

  public console?: String;

  public year?: DateTime;

  public notes?: String;

  public completed?: Boolean;

  constructor(
    id: String,
    title: String,
    completionDate: DateTime,
    console: String,
    year: DateTime,
    notes: String,
    completed: Boolean,
  ) {
    this.id = id;
    this.title = title;
    this.completionDate = completionDate;
    this.console = console;
    this.year = year;
    this.notes = notes;
    this.completed = completed;
  }
}

export default Game;
